cpp/src/cider/benchmarks/HashMapBenchmark.cpp

```diff
@@ -0,0 +1,165 @@
+/*
+ * Copyright(c) 2022-2023 Intel Corporation.
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+#include <unistd.h>
+#include <iomanip>
+#include <iostream>
+#include <random>
+#include <vector>
+
+#include <benchmark/benchmark.h>
+#include <common/hashtable/FixedHashMap.h>
+#include <common/hashtable/HashMap.h>
+#include <common/hashtable/HashTable.h>
+#include <common/hashtable/HashTableAllocator.h>
+#include <tests/utils/Utils.h>
+
+using namespace cider::hashtable;
+
+enum GeneratePattern { Sequence, Random };
+
+#define DBMS_HASH_MAP_COUNT_COLLISIONS
+#define DBMS_HASH_MAP_DEBUG_RESIZES
+
+#define N_MAX std::numeric_limits<T>::max()
+#define N_MIN std::numeric_limits<T>::min()
+
+#define NO_INLINE __attribute__((__noinline__))
+
+template <typename KeyType, typename Map>
+int32_t NO_INLINE bench(const std::vector<KeyType>& data, const char* name) {
+  Map map;

```

Comment:
For this function, `Map` should be a class with `KeyType` parameters? 
how to avoid such case?
```
using  KeyType1 = uint8_t;
using  KeyType2 = uint16_t;
auto ret = bench<KeyType1, xxHashMap<KeyType2>>(xx,xx);
```

Reply:
Moved the declaration `BaselineLookup` and `OptimizedLookup` into benchmark functions, so that we can use `KeyType` of template to define them.

Btw, removed parameter `name` since it's not in use.